src/Fetcher/TululaFetcher.php

```diff
@@ -0,0 +1,241 @@
+use Symfony\Component\String\Slugger\AsciiSlugger;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class TululaFetcher implements FetcherInterface
+{
+    const SOURCE = 'tulula';
+    const TAGS_SYNONYMS = [
+        'android',
+        'css',
+        'data',
+        'devops',
+        'dotnet',
+        'facebook',
+        'golang',
+        'html',
+        'ios',
+        'java',
+        'javascript',
+        'nodejs',
+        'php',
+        'python',
+        'react native',
+        'ruby',
+        'rust',
+        'security',
+        'ux',
+        'ai/ml',
+        'sql',
+        'iot',
+    ];
+
+    private $em;
+    private $logger;
+    private $tagRepository;
+    private $excludedTags;
+    private $locationGuesser;
+    private $slugger;
+    private HttpClientInterface $client;
+
+    public function __construct(ManagerRegistry $doctrine, LoggerInterface $logger, LocationGuesser $locationGuesser, HttpClientInterface $client)
+    {
+        $this->client = $client;
+        $this->em = $doctrine->getManager();
+        $this->logger = $logger;
+        $this->tagRepository = $this->em->getRepository(Tag::class);
+        $this->excludedTags = $this->em->getRepository(ExcludedTag::class)->findAll();
+        $this->locationGuesser = $locationGuesser;
+        $this->slugger = new AsciiSlugger();
+    }
+
+    public function isActive(): bool
+    {
+        return true;
+    }
+
+    public function getUrl(array $params = []): string
+    {
+        return 'https://tulu.la/api/public';
+    }
+
+    public function fetch(): \Generator
+    {
+        foreach ($this->tagRepository->findBy(['selected' => true]) as $tag) {
+            $tagKey = array_search($this->slugger->slug(strtolower($tag->getName()))->toString(), self::TAGS_SYNONYMS);
+
+            if (false === $tagKey) {
+                continue;
+            }
+
+            $tagSynonym = self::TAGS_SYNONYMS[$tagKey];
+            $sameTagConferences = [];
+            foreach ($this->queryTululaApi() as $conference) {
+                foreach ($conference['tags'] as $conferenceTag) {
+                    if ($tagSynonym === strtolower($conferenceTag['name'])) {
+                        $sameTagConferences[] = $conference;
+                        break;
+                    }
+                }
+            }
+
+            yield $this->getUrl() => $this->denormalizeConferences($sameTagConferences, $tag);
+        }
+    }
+
+    public function denormalizeConferences(array $rawConferences, Tag $tag): array
+    {
+        $conferences = [];
+        foreach ($rawConferences as $rawConference) {
+            if (!$rawConference['isOnline']) {
+                // There are cases where Tulula doesn't provide city or state so the string might be empty
+                if (!empty($city = $rawConference['venue']['city'] ?: $rawConference['venue']['state'])) {
```

Comment:
This code is not really easy to debug since `empty` is a catch all.
It should use:
```suggestion
                if ($city = $rawConference['venue']['city'] ?? $rawConference['venue']['state'] ?? false) {
```